server/post/model.ts

```diff
@@ -0,0 +1,51 @@
+import type { Types } from "mongoose";
+import { Schema, model } from "mongoose";
+import { User } from "server/user/model";
+
+export type Post = {
+  _id: Types.ObjectId;
+  authorId: Types.ObjectId;
+  lectureId: Types.ObjectId;
+  parentId: Types.ObjectId;
+  content: string;
+  dateCreated: Date;
+};
+
+export type PopulatedPost = {
+  _id: Types.ObjectId;
+  authorId: User;
+  lectureId: Types.ObjectId;
```

Comment:
This'll doesn't need to be made into Lecture object later right? I guess there isn't really any lecture specific info we need for a post other than which lecture it belongs to